python/lsst/meas/algorithms/subtractBackground.py

```diff
@@ -0,0 +1,260 @@
+            "MEAN": "unclipped mean",
+            "MEDIAN": "median",
+            }
+    )
+    undersampleStyle = pexConfig.ChoiceField(
+        doc="behaviour if there are too few points in grid for requested interpolation style",
+        dtype=str, default="REDUCE_INTERP_ORDER",
+        allowed={
+            "THROW_EXCEPTION": "throw an exception if there are too few points",
+            "REDUCE_INTERP_ORDER": "use an interpolation style with a lower order.",
+            "INCREASE_NXNYSAMPLE": "Increase the number of samples used to make the interpolation grid.",
+            },
+    )
+    binSize = pexConfig.RangeField(
+        doc="how large a region of the sky should be used for each background point",
+        dtype=int, default=256, min=10,
+    )
+    algorithm = pexConfig.ChoiceField(
+        doc="how to interpolate the background values. This maps to an enum; see afw::math::Background",
+        dtype=str, default="NATURAL_SPLINE", optional=True,
+        allowed={
+            "CONSTANT" : "Use a single constant value",
+            "LINEAR" : "Use linear interpolation",
+            "NATURAL_SPLINE" : "cubic spline with zero second derivative at endpoints",
+            "AKIMA_SPLINE": "higher-level nonlinear spline that is more robust to outliers",
+            "NONE": "No background estimation is to be attempted",
+            },
+    )
+    ignoredPixelMask = pexConfig.ListField(
+        doc="Names of mask planes to ignore while estimating the background",
+        dtype=str, default = ["BAD", "EDGE", "DETECTED", "DETECTED_NEGATIVE", "NO_DATA",],
+        itemCheck = lambda x: x in afwImage.MaskU().getMaskPlaneDict().keys(),
+    )
+    isNanSafe = pexConfig.Field(
+        doc="Ignore NaNs when estimating the background",
+        dtype=bool, default=False,
+    )
+
+    useApprox = pexConfig.Field(
+        doc="Use Approximate (Chebyshev) to model background.",
+        dtype=bool, default=False,
+    )
+    approxOrderX = pexConfig.Field(
+        doc="Approximation order in X for background Chebyshev (valid only with useApprox=True)",
+        dtype=int, default=6,
+    )
+    # Note: Currently X- and Y-orders must be equal due to a limitation in math::Chebyshev1Function2
+    # The following is being added so that the weighting attribute can also be configurable for the
+    # call to afwMath.ApproximateControl
+    approxOrderY = pexConfig.Field(
+        doc="Approximation order in Y for background Chebyshev (valid only with useApprox=True)",
+        dtype=int, default=-1,
+    )
+    weighting = pexConfig.Field(
+        doc="Use inverse variance weighting in calculation (valid only with useApprox=True)",
+        dtype=bool, default=True,
+    )
+
+    def validate(self):
+        pexConfig.Config.validate(self)
+        # Allow None to be used as an equivalent for "NONE", even though C++ expects the latter.
+        if self.algorithm is None:
+            self.algorithm = "NONE"
```

Comment:
I think we always validate before freezing, but it is still disturbing to be changing things in validate. I was preserving existing behavior, but I don't like it.